src/content/docs/reference/test/index.mdx

```diff
@@ -0,0 +1,264 @@
+
+1. Project defaults (`test.yaml` files, applied per directory).
+2. Per-test frontmatter (YAML for `.tql`/`.xxd`, `# key: value` comments for
+   Python files).
+
+Common frontmatter keys:
+
+| Key        | Type            | Default   | Description                                            |
+| ---------- | --------------- | --------- | ------------------------------------------------------ |
+| `runner`   | string          | by suffix | Runner name (`tenzir`, `python`, custom).              |
+| `fixtures` | list of strings | `[]`      | Requested fixtures; use `fixture` for a single value.  |
+| `timeout`  | integer (s)     | `30`      | Command timeout. (`--coverage` multiplies it by five.) |
+| `error`    | boolean         | `false`   | Expect a non-zero exit code.                           |
+| `skip`     | string          | unset     | Mark the test as skipped (reason required).            |
+
+`test.yaml` files accept the same keys and apply recursively to child
+directories. Adjacent `tenzir.yaml` files still configure the Tenzir binary; the
+harness appends `--config=<file>` automatically.
+
+## Fixtures
+
+### Declaring fixtures
+
+- List fixture names in frontmatter (`fixtures: [node, http]`).
+- The harness encodes requests in `TENZIR_TEST_FIXTURES` and exposes helper APIs
+  in `tenzir_test.fixtures`:
+  - `requested()` – Read-only view of active fixtures.
+  - `require("name")` – Assert that a fixture was requested.
+  - `Executor()` – Convenience wrapper that runs Tenzir commands with resolved
+    binaries and timeout budget.
+
+Example use from a Python helper:
+
+```python
+from tenzir_test.fixtures import Executor
+
+executor = Executor()
+result = executor.run("from_file 'inputs/events.ndjson' | where severity >= 5\n")
+assert result.returncode == 0
+```
+
+### Registering fixtures
+
+Implement fixtures in `fixtures/` and register them with decorators:
+
+- `@tenzir_test.startup()` (or `@startup(replace=True)`) – Run before the test.
+  Return a mapping of environment variables for the test process.
+- `@tenzir_test.teardown()` – Receive the environment mapping and perform
+  cleanup.
+- Fixtures can also return `FixtureHandle` objects to bundle teardown logic.
+
+The [fixture guide](/guides/testing/create-fixtures) demonstrates an HTTP echo
+server that exposes `HTTP_FIXTURE_URL` and tears down cleanly. When your fixture
+needs the built-in node, call `fixtures.require("node")` to fail fast.
+
+### Built-in `node` fixture
+
+- Available as `fixtures: [node]`.
+- Starts a transient `tenzir-node`, exposes:
+  - `TENZIR_NODE_CLIENT_BINARY`
+  - `TENZIR_NODE_CLIENT_ENDPOINT`
+  - `TENZIR_NODE_CLIENT_TIMEOUT`
+- Pipelines launched by the `tenzir` runner automatically receive
+  `--endpoint=<value>` when the fixture is active.
+
+### Custom shell fixtures
+
+Place scripts in `custom/` (with `.sh`) to run them under `bash -eu`. During
+execution the harness sets:
```

Comment:
Is that so? Didn't fully check the revised framework, but does the directory matter? Also, the whole `custom` thing was never really finished and tested. Not sure whether you did.